test/inductor/test_fp8.py

```diff
@@ -98,6 +121,183 @@ def fp8_cast(x, dtype):
             x = torch.rand(*x_shape, device="cuda").to(dtype=torch.float8_e5m2)
             y = compiled_fp8_cast(x, torch.float8_e4m3fn)
 
+    @unittest.skipIf(TEST_WITH_ROCM, "FP8 is not supported on ROCM")
+    @unittest.skipIf(not SM90OrLater, "FP8 is only supported on H100+")
+    @parametrize("src_dtype", (torch.float16, torch.bfloat16, torch.float))
+    @parametrize("dst_dtype", (torch.float8_e4m3fn, torch.float8_e5m2))
+    @parametrize("shape", ((16, 16, 16), (4, 2048, 4096)))
+    def test_to_fp8_saturated(
+        self, src_dtype: torch.dtype, dst_dtype: torch.dtype, shape: Tuple[int]
+    ):
+        def fp8_saturated(x, dtype):
+            return _to_fp8_saturated(x, dtype)
+
+        compiled_fp8_cast = torch.compile(
+            fp8_saturated, backend="inductor", dynamic=True
+        )
+        x = torch.rand(*shape, device="cuda", dtype=src_dtype)
+        y_compiled = compiled_fp8_cast(x, dst_dtype)
+        y = fp8_saturated(x, dst_dtype)
+
+        torch.testing.assert_close(y_compiled.half(), y.half(), rtol=5e-1, atol=5e-1)
+
+    @unittest.skipIf(TEST_WITH_ROCM, "FP8 is not supported on ROCM")
+    @unittest.skipIf(not SM90OrLater, "FP8 is only supported on H100+")
+    @parametrize("float8_dtype", (torch.float8_e4m3fn, torch.float8_e5m2))
+    @parametrize(
+        "shape", ((1, 1, 15), (1, 10, 15), (1, 10, 512), (1, 10, 4096), (4, 2048, 4096))
+    )
+    def test_amax_fp8_quant(self, float8_dtype: torch.dtype, shape: Tuple[int]):
+        batch_size, sequence_length, hidden_size = shape
+
+        def amax_fp8(x: Tensor, scale: Tensor):
+            y = torch.max(torch.abs(x))
```

Comment:
should this be `amax` instead of `max`? If `max` is not intentional, I think using `amax` to mean "return the values without indices" is clearer